src/windows/ride.c

```diff
@@ -4992,17 +4996,21 @@ static void window_ride_measurements_paint(rct_window *w, rct_drawpixelinfo *dpi
 					length = ride->length[numLengths];
 					if (length != 0) {
 						length >>= 16;
-						set_format_arg(0 + (numLengths * 4), uint16, 1346);
+						set_format_arg(0 + (numLengths * 4), uint16, STR_RIDE_LENGTH_ENTRY_WITH_SEPARATOR);
 						set_format_arg(2 + (numLengths * 4), uint16, (length & 0xFFFF));
 						numLengths++;
 					}
 				}
 				if (numLengths == 0) {
-					set_format_arg(0, uint16, 1346);
+					set_format_arg(0, uint16, STR_RIDE_LENGTH_ENTRY);
 					set_format_arg(2, uint16, 0);
 					numLengths++;
+				} else {
+					//sadly, STR_RIDE_LENGTH_ENTRY_WITH_SEPARATOR are defined with the separator AFTER an entry
+					//therefore we set the last entry to use the no-separator format now, post-format
+					set_format_arg(0 + ((numTimes - 1) * 4), uint16, STR_RIDE_LENGTH_ENTRY);
```

Comment:
It definitely should. Thanks for that. I already commited a fix and pushed to the branch used in this PR .. but I guess due to this PR being merged already, it won't show up here. Should I open a new PR then?